src/cider/nrepl/middleware/test.clj

```diff
@@ -300,29 +272,42 @@
                              ~@body)
                            (alter-meta! session# dissoc :thread :eval-msg))))))
 
-(defn handle-test-op
-  [{:keys [ns tests session transport include exclude] :as msg}]
-  (with-interruptible-eval msg
-    (if-let [ns (ns/ensure-namespace ns)]
-      (let [nss {ns (map u/as-sym tests)}
-            report (test-nss nss include exclude)]
+(defn handle-test-var-query-op
+  [{:keys [var-query transport] :as msg}]
+  (with-interruptible-eval
+    msg
+    (try
+      (let [report (test-var-query
+                    (-> var-query
+                         ;; TODO: Should this be done, or should the client specify it?
```

Comment:
@bbatsov is this TODO a yes or no? Happy to remove it / act upon it.

Reply:
Why would the client want to run tests in a ns without tests?

Reply:
I don't think it would make sense to, but it wouldn't cause errors either. I can't see any harm in this right now. I shall remove the TODO.